url.go

```diff
@@ -151,18 +152,35 @@ func ParseReference(url string) (*Reference, error) {
 }
 
 func parseReference(url string) (*Reference, error) {
```

Comment:
Please update the doc comment on ParseReference to include the new forms of URL.